ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -33,7 +42,54 @@ public IngestionReportMethod getReportMethod() {
         return reportMethod;
     }
 
-    public Map<String, String> getAdditionalProperties() {
+    public ArrayList<String> getDropByTags() {
+        return dropByTags;
+    }
+
+    public ArrayList<String> getIngestByTags() {
+        return ingestByTags;
+    }
+
+    public ArrayList<String> getAdditionalTags() {
+        return additionalTags;
+    }
+
+    public ArrayList<String> getIngestIfNotExists() {
+        return ingestIfNotExists;
+    }
+
+    public Map<String, String> getAdditionalProperties() throws IOException {
+        if(!dropByTags.isEmpty() || !ingestByTags.isEmpty() || !additionalTags.isEmpty()){
```

Comment:
The value is kept in additionalProperties even between get calls. What will happen if all those are empty - looks like the old "tags" will still be there, right?

Reply:
will return a new object
Its actually very weird the way it works now - javadoc explains that all is added to additionalProperties which is private and shouldn't be known to user ...